libpromises/pipes_unix.c

```diff
@@ -39,7 +39,7 @@ static bool CfSetuid(uid_t uid, gid_t gid);
 static int cf_pwait(pid_t pid);
 
 static pid_t *CHILDREN = NULL; /* GLOBAL_X */
-static int MAX_FD = 128; /* GLOBAL_X */ /* Max number of simultaneous pipes */
+static int MAX_FD = 8192; /* GLOBAL_X */ /* Max number of simultaneous pipes */
```

Comment:
Wow, big hammer approach. :) I'm not quite sure 8192 child processes (or even 4096) running at the same time is feasible. What about 2048?

Reply:
:) yeah, maybe a little bit overstated, i like to live dangerously. 2048 should be enough.

Reply:
I was going to stuffiest 640k

Reply:
hmm...basically i can patch this locally, as i am constantly building cfengine rpms for aix.  https://github.com/cfengine/libntech/pull/130 looks very "promising" to me, and i dont want to fire up a "numbers" flamewar :-). this problem seems to be a isolated thing (only me) anyway. nobody is doing stupid stuff like me *g*

Reply:
128 is definitely a very low and totally artificial limit. 2048 should be fine for everybody. :wink: